File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.cpp
#include <algorithm>
#include <queue>
#include <vector>

using namespace std;

int solution(vector<int> scoville, int K) {
  int answer = 0;
  priority_queue<int, vector<int>, greater<int>> scovilleq;
  for (int i = 0; i < scoville.size(); i++)
    scovilleq.push(scoville[i]);
  while (scovilleq.top() < K) {
    int a, b;
    if (scovilleq.size() == 1)
      return -1;
    a = scovilleq.top();
    scovilleq.pop();
    b = scovilleq.top();
    scovilleq.pop();
    scovilleq.push(a + b * 2);
    answer++;
  }
  return answer;
}

File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    heap=[]
    answer=0
    heapq.heapify(scoville)
    while len(scoville)>1:
        a=heapq.heappop(scoville)
        b=heapq.heappop(scoville)
        if a<K or b<K:
            heapq.heappush(scoville, a+b*2)
            answer=answer+1
        else:
            return answer
    if scoville[0]>K:
        return answer
    else:
        return -1